node/packages/client/src/ethereum/parser/foxy.ts

```diff
@@ -0,0 +1,69 @@
+import { Tx as BlockbookTx } from '@shapeshiftoss/blockbook'
+import { ethers } from 'ethers'
+import { TxParser } from '../../types'
+import { TxSpecific, SubParser } from '../types'
+import { getSigHash, txInteractsWithContract } from './utils'
+import { FOXY_STAKING_CONTRACT } from './constants'
+import FOXY_STAKING_ABI from './abi/foxyStaking'
+
+export class Parser implements SubParser {
+  abiInterface: ethers.utils.Interface
+
+  readonly stakeSigHash: string
+  readonly unstakeSigHash: string
+  readonly instantUnstakeSigHash: string
+  readonly claimWithdrawSigHash: string
+
+  constructor() {
+    this.abiInterface = new ethers.utils.Interface(FOXY_STAKING_ABI)
+    // Hard coded staking sigHash because there is two different stake methods
+    // in the foxy staking contract. This gives us the ability to parser the correct one.
+    ;(this.stakeSigHash = '0x7acb7757'),
+      (this.unstakeSigHash = this.abiInterface.getSighash('unstake')),
+      (this.instantUnstakeSigHash = this.abiInterface.getSighash('instantUnstake')),
+      (this.claimWithdrawSigHash = this.abiInterface.getSighash('claimWithdraw'))
+  }
+
+  async parse(tx: BlockbookTx): Promise<TxSpecific | undefined> {
+    if (!txInteractsWithContract(tx, FOXY_STAKING_CONTRACT)) return
+    const txData = tx.ethereumSpecific?.data
+    if (!txData) return
+
+    const txSigHash = getSigHash(txData)
+    const abiInterface = this.getAbiInterface(txSigHash)
```

Comment:
This feels a little misleading - we already have an `abiInterface` on the class. What this is actually doing is checking to see if the `this.abiInterface` supports any of the supported foxy functions.

Can we rename `getAbiInterface` to something like `abiInterfaceSupportsSigHash`, and then early return on line 34 if not?

Then we could use the interface on our class instead of creating a second one.